infra/modules/azure_monorepo_single_env_starter_pack/github_repository.tf

```diff
@@ -0,0 +1,41 @@
+resource "github_repository" "this" {
+  name        = var.repository.name
+  description = var.repository.description
+  topics      = var.repository.topics
+
+  visibility = "public"
+
+  allow_auto_merge            = true
+  allow_rebase_merge          = false
+  allow_merge_commit          = false
+  allow_squash_merge          = true
+  squash_merge_commit_title   = "PR_TITLE"
+  squash_merge_commit_message = "BLANK"
+
+  delete_branch_on_merge = true
+
+  has_projects    = false
+  has_wiki        = false
+  has_discussions = false
+  has_issues      = false
+  has_downloads   = false
+
+  vulnerability_alerts = true
+
+  archive_on_destroy  = true
+  allow_update_branch = true
+
+  security_and_analysis {
+    secret_scanning {
+      status = "enabled"
+    }
+
+    secret_scanning_push_protection {
+      status = "enabled"
+    }
+
+    advanced_security {
+      status = "enabled"
+    }
+  }
```

Comment:
we should add a validation to set required tags (team name / domain)

Reply:
Can you explain it a little more?

Reply:
I mean: tags shouldn't ever be an empty map, but contains at least the team / domain.

Reply:
Which tags? I don't follow you :)

Reply:
the usual ones `Owner` , `Source`, ... aren't required?